shelley/chain-and-ledger/formal-spec/properties.tex

```diff
@@ -1,10 +1,312 @@
+  Notice that $\var{unclaimed}$ is added to $\var{treasury}$ and subtracted from the $\var{deposits}$.
+  Moreover, $\var{refunded}$ is subtracted from $\var{deposits}$.
+  (We will prove that $\var{deposits}-\var{unclaimed}+\var{refunded}$ is non-negative in a subsequent theorem.)
+  It therefore suffices to show that
+  \begin{equation*}
+    \begin{array}{r@{~=~}l}
+    \Val(\var{rewards}\unionoverridePlus\var{refunds})
+    & \Val(\var{rewards}) + \Val(\var{refunds})
+    \\
+    & \Val(\var{rewards}) + \var{refunded}
+    \end{array}
+  \end{equation*}
+  But this is clear from the definition of $\unionoverridePlus$.
+\end{proof}
+
+\begin{lemma}
+  \label{lemma:ru-pres-of-value}
+  For all mappings $b$ of blocks made, epochs $\epsilon$, and epoch states $s$,
+  \begin{equation*}
+    \Val(s) = \Val(\fun{applyRUpd}~(\fun{createRUpd}~b~s)~\epsilon~s)
+  \end{equation*}
+\end{lemma}
+
+\begin{proof}
+  In the definition of $\fun{applyRUpd}$ in Figure~\ref{fig:functions:reward-update-application},
+  we see that:
+  \begin{equation*}
+      \Delta t + \Delta r + \Val(rs) + \Val(\var{update_{rwd}}) + \Delta d + \Delta f = 0
+  \end{equation*}
+  In the definition of $\fun{createRUpd}$ in Figure~\ref{fig:functions:reward-update-creation},
+  we see that:
+  \begin{equation*}
+    \begin{array}{r@{~=~}l}
+      \var{rewardPot} & \var{feeSS} + \Delta r_1 \\
+      \var{R} & \var{rewardPot} - \Delta t_1 \\
+      \Delta t_2 & R \\
+    \end{array}
+  \end{equation*}
+  Therefore
+  \begin{equation*}
+    \begin{array}{c}
+      (\var{feeSS} + \Delta r_1) - \Delta t_1 - \Val(rs) = \Delta t_2 \\
+      0 = (\Delta t_1 + \Delta t_2 ) - \Delta r_1 + \Val(rs)- \var{feeSS} \\
+    \end{array}
+  \end{equation*}
+  So it suffices to show that $\Delta d + \Val(\var{update_{rwd}}) = \var{rewards}_{\var{mir}}$.
+  Notice that
+  \begin{equation*}
+    \begin{array}{r@{~=~}l}
+      \Val(\var{update}_{rwd}) & \Val(registered) + \Val(newlyRegistered')
+      \\
+      \var{rewvards}_{mir} & \Val(registered) + \Val(newlyRegistered)
+      \\
+      \Delta d + \Val(newlyRegistered') & \Val(newlyRegistered)
+    \end{array}
+  \end{equation*}
+  Therefore $\Delta d + \Val(\var{update_{rwd}}) = \var{rewards}_{\var{mir}}$
+  follows by adding $\Val(registered)$ to both sides of the third
+  equality above, and then substituting with the top two.
+\end{proof}
+
+We are now ready to prove Theorem~\ref{thm:chain-pres-of-value}.
+
+\begin{proof}
+  For a given transition $\mathsf{TR}$, let \POV{TR}
+  be the statement:
+
+  for all environments $e$, signals $\sigma$, and states $s$, $s'$,
+  \begin{equation*}
+    \text{if }e\vdash s\trans{tr}{\sigma}s'\text{ then }\Val(s) = \Val(s')
+  \end{equation*}
+  Our goal is to prove \POV{CHAIN}.
+  Lemmas~\ref{lemma:utxo-pres-of-value} and \ref{lemma:delegs-pres-of-value} imply \POV{LEDGER},
+  since $\mathsf{UTXOW}$ transforms state exactly as $\mathsf{UTXOW}$ does.
```

Comment:
as UTXO does ?

Reply:
yep, I pushed some changes from @kevinhammond 's review comments, which included this. thanks!